client/lib/domains/whois/reducer.js

```diff
@@ -63,6 +65,10 @@ function reducer( state, payload ) {
 		case ActionTypes.WHOIS_UPDATE_COMPLETED:
 			state = updateDomainState( state, action.domainName, {
 				needsUpdate: true,
+				registrantContactDetails: {
```

Comment:
Why introduce a new field instead of merging it into the appropriate one in `data`? This is ripe for introducing a bug in the future.
Much less changes needed too. Or did I miss something?

Reply:
Thanks @klimeryk
 
Good point. I initially went down this path, but thought it might be cleaner to have a model explicitly for the form, whose props differ from the whois data object. I'll put those changes back in and test it out.